opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/validation/Xs2aValidator.java

```diff
@@ -56,6 +67,15 @@ public void validate(DelegateExecution exec, Object... dtosToValidate) {
         );
     }
 
+    private Set<FieldCode> getFieldsToIgnoreValidate(Xs2aContext context, Class invokerClass) {
+        Approach approach = context.aspspProfile().getPreferredApproach();
+        return context.getRequestScoped().getValidationRules().stream()
+                .filter(it -> !EMBEDDED.equals(approach) || it.isForEmbedded())
```

Comment:
For readability, it is better to use straight logic and invert it afterward, i.e.:
`!(EMBEDDED.equals(approach) && it.isForEmbedded())`